src/main/resources/com/google/api/codegen/go/main.snip

```diff
@@ -81,35 +81,35 @@
         }
     }
 
-    @join retryDef : context.entrySet(context.getApiConfig.getInterfaceConfig(service).getRetrySettingsDefinition)
-        func {@context.lowerUnderscoreToLowerCamel(retryDef.getKey)}{@context.getClientPrefix(service)}RetryOptions() []gax.CallOption {
-            return []gax.CallOption{
-                gax.WithTimeout({@retryDef.getValue.getTotalTimeout.getMillis()}*time.Millisecond),
-                gax.WithDelayTimeoutSettings({@retryDef.getValue.getInitialRetryDelay.getMillis}*time.Millisecond, {@retryDef.getValue.getMaxRetryDelay.getMillis}*time.Millisecond, {@retryDef.getValue.getRetryDelayMultiplier}),
-                gax.WithRPCTimeoutSettings({@retryDef.getValue.getInitialRpcTimeout.getMillis}*time.Millisecond, {@retryDef.getValue.getMaxRpcTimeout.getMillis}*time.Millisecond, {@retryDef.getValue.getRpcTimeoutMultiplier}),
-            }
-        }
-    @end
-
     func default{@context.getClientPrefix(service)}CallOptions() *{@context.getClientPrefix(service)}CallOptions {
-        @join retryCodesDef : context.entrySet(context.getApiConfig.getInterfaceConfig(service).getRetryCodesDefinition) if retryCodesDef.getValue
-            with{@context.lowerUnderscoreToUpperCamel(retryCodesDef.getKey)}RetryCodes := gax.WithRetryCodes([]codes.Code{
-                @join code : retryCodesDef.getValue
-                    codes.{@context.upperUnderscoreToUpperCamel(code.toString())},
+        retry := map[[2]string][]gax.CallOption{
+            @join retryName : context.getRetryConfigNames(service) if context.getApiConfig.getInterfaceConfig(service).getRetryCodesDefinition.get(retryName.getCodesName)
```

Comment:
optional: assign `interfaceConfig` or `retryCodesDefinition` through `@let`? The repetition of `context.getApiConfig...` isn't quite readable.